src/main/java/net/openhft/chronicle/core/pool/ClassAliasPool.java

```diff
@@ -146,22 +148,39 @@ public Class<?> forName(@NotNull CharSequence name) throws ClassNotFoundRuntimeE
         Class<?> clazz = stringClassMap.get(key);
         if (clazz != null)
             return clazz;
-        clazz = stringClassMap2.get(key);
-        if (clazz != null) return clazz;
         return forName0(key);
     }
 
     @NotNull
     private synchronized Class<?> forName0(@NotNull CAPKey key) throws ClassNotFoundRuntimeException {
         Class<?> clazz = stringClassMap2.get(key);
-        if (clazz != null) return clazz;
+        if (clazz != null) {
+            if (clazz == ResolutionFailure.class)
+                throw cachedEx;
```

Comment:
shouldn't we throw a new exception with the class details filled in?

Reply:
Yeah I think having a class name in there would make troubleshooting a lot easier if it needed to be done. Perhaps it would mean a lot of object creation with all the exceptions. I suppose you could cache those too ;)

Reply:
I have a better idea that I will implement, thank you for the food for thoughs.